executor/mem_reader.go

```diff
@@ -82,18 +81,12 @@ func (m *memIndexReader) getMemRows() ([][]types.Datum, error) {
 		tps = append(tps, types.NewFieldType(mysql.TypeLonglong))
 	}
 
-	mutableRow := chunk.MutRowFromTypes(m.retFieldTypes)
 	err := iterTxnMemBuffer(m.ctx, m.kvRanges, func(key, value []byte) error {
 		data, err := m.decodeIndexKeyValue(key, value, tps)
 		if err != nil {
 			return err
 		}
 
-		mutableRow.SetDatums(data...)
-		matched, _, err := expression.EvalBool(m.ctx, m.conditions, mutableRow.ToRow())
```

Comment:
Move those condition check to `UnionScanExec.Next` may have a performance regression? 
Why not `EvalVirtualColumn` in `UnionScanExec.open` for memReader?